src/rollup/types.d.ts

```diff
@@ -316,7 +316,13 @@ export type HasModuleSideEffects = (id: string, external: boolean) => boolean;
 
 export type LoadResult = SourceDescription | string | NullValue;
 
-export type LoadHook = (this: PluginContext, id: string) => LoadResult;
+export type LoadHook = (
+	this: PluginContext,
+	id: string,
+	options: {
+		attributes: Record<string, any>;
```

Comment:
I'm not sure if we should update the `attributes` type to `type Attributes = Record<string, string | Attributes>` right now. Currently, it looks like estree still considers the value to be `string` https://github.com/estree/estree/blob/master/es2025.md. Maybe we can keep things as they are and see how it evolves. By the way, I find the types in tc39 a bit hard to understand 😢.

Reply:
I agree with @TrickyPi here. Also, I find the spec rather ambiguous here. Is this to mean it could look like this?

```
with { foo: {bar: 'baz'} }
```

But then, I would have expected the braces to be part of WithEntries or similar. Considering that all relevant cases at the moment are flat objects, I would hope we can stick with this for now, as this would also keep the refactoring to a reasonable amount within Rollup.

Reply:
Oh, my bad, it seems `Record<string, string>` is correct.
https://github.com/tc39/proposal-import-attributes?tab=readme-ov-file#should-more-than-just-strings-be-supported-as-attribute-values
I had the old version of the spec in my mind and mistakenly verified that 🤦.